arangod/Aql/OptimizerRules.cpp

```diff
@@ -7471,6 +7470,29 @@ void arangodb::aql::geoIndexRule(Optimizer* opt,
     LimitNode* limit = nullptr;
     bool canUseSortLimit = true;
 
+    bool mustRespectIdxHint = false;
+    if (plan->hasForcedIndexHints()) {
+      auto enumerateColNode =
+          ExecutionNode::castTo<EnumerateCollectionNode const*>(node);
+      auto indexes = enumerateColNode->collection()->indexes();
+      auto idxNames = enumerateColNode->hint().hint();
+      for (auto const& idxName : idxNames) {
+        for (std::shared_ptr<Index> idx : indexes) {
+          if (idx->name() == idxName) {
+            auto idxType = idx->type();
+            if ((idxType != Index::IndexType::TRI_IDX_TYPE_GEO1_INDEX) &&
+                (idxType != Index::IndexType::TRI_IDX_TYPE_GEO2_INDEX) &&
+                (idxType != Index::IndexType::TRI_IDX_TYPE_GEO_INDEX)) {
+              mustRespectIdxHint = true;
```

Comment:
Looks like we could "break" here. As we already know that geo optimization is not an option anymore